jekyll/_cci2/pipeline-parameters.md

```diff
@@ -0,0 +1,195 @@
+
+
+## Config Processing Stages and Parameter Scopes
+
+### Processing Stages
+
+Configuration processing happens in the following stages:
+
+- Pipeline parameters are resolved and type-checked
+- Pipeline parameters are replaced in the orb statement
+- Orbs are imported
+
+The remaining configuration is processed, element parameters are resolved, type-checked, and substituted,
+
+## Element Parameter Scope
+
+Element parameters use lexical scoping, so parameters are in scope within the element they are defined in, e.g. a job, a command, or an executor. If an element with parameters calls another element with parameters, like in the example below, the inner element does not inherit the scope of the calling element.
+
+```
+version: 2.1
+
+commands:
+    print:
+      parameters:
+          message:
+          type: string
+      steps:
+          - run: echo << parameters.message >>
+
+jobs:
+    cat-file:
+      parameters:
+          file:
+          type: string
+      steps:
+          - print:
+              message: Printing << parameters.file >>
+          - run: cat << parameters.file >>
+
+workflows:
+    my-workflow:
+      jobs:
+          - cat-file:
+              file: test.txt
+```
+
+Even though the `print` command is called from the cat-file job, the file parameter would not be in scope inside the print. This ensures that all parameters are always bound to a valid value, and the set of available parameters is always known.
+
+## Pipeline Value Scope
+
+Pipeline values, the pipeline-wide values that are provided by CircleCI (e.g. << pipeline.number >>) are always in scope.
+
+### Pipeline Parameter Scope
+
+Pipeline parameters which are defined in configuration are always in scope, with two exceptions:
+
+- Pipeline parameters are not in scope for the definition of other pipeline parameters, so they cannot depend on one another.
+- Pipeline parameters are not in scope in the body of orbs, even inline orbs, to prevent data leaks.
+
+## Conditional Workflows
+
+New as of June 2019, you can use a when clause (we also support the inverse clause unless) under a workflow declaration with a boolean value to decide whether or not to run that workflow.
+
+The most common use of this construct is to use a pipeline parameter as the value, allowing an API trigger to pass that parameter to determine which workflows to run.
+
+Below is an example configuration using two different pipeline parameters, one used to drive whether a particular workflow will run and another to determine if a particular step will run.
+
+```
+version: 2.1
+
+parameters:
+  run_integration_tests:
+    type: boolean
+    default: false
+  deploy:
+    type: boolean
+    default: false
+
+workflows:
+  version: 2
+  integration_tests:
+    when: << pipeline.parameters.run_integration_tests >>
+    jobs:
+      - mytestjob
+      - when:
+          condition: << pipeline.parameters.deploy >>
+          steps:
+            - deploy
```

Comment:
I think that was the faulty example that was just removed from the docs, the `when` with `steps` is not valid in this context afaik.

I just fixed this at [the upstream](https://github.com/CircleCI-Public/api-preview-docs/blob/master/docs/conditional-workflows.md).

Reply:
copied in the example, thanks! 🙏